### PR TITLE
fix(Accordion): updates layout

### PR DESCRIPTION
**Pull request by rapahaeru**
## Description
https://jirasoftware.catho.com.br/browse/QTM-260
figma: https://www.figma.com/file/SED32ByxEn8C1HR5J5aR2JGs/Style-Guide---Design-System?node-id=11761%3A22263

previously
image

now:
image

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [ ] TypeScript updated
- [ ] a11y

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile
- [X] Ux review validation

### CODE REVIEW

File: components/Accordion/Accordion.jsx [hunk 1]
```diff
@@ -20,67 +21,82 @@ const Wrapper = styled.ul`
...
 
-const AccordionItem = styled.li`
-  border-bottom: 1px solid
-    ${({ opened, color }) => (opened ? 'transparent' : color)};
-
-  &:last-child {
-    border-bottom: 1px solid transparent;
-  }
-`;
+const AccordionItem = styled.li``;
```
Comment (tcelestino) [inline components/Accordion/Accordion.jsx:32]: is it need this?

**Review by tcelestino — CHANGES_REQUESTED** [review]

File: components/Accordion/Accordion.jsx (hunk 1 above)
Comment (rapahaeru) [reply components/Accordion/Accordion.jsx:32]: AccordionItem is needed to find it self in unit test.
If i remove it, will be necessary to put a class in html renderized and find in unit test by class ( in css tree way).
I believe keeping in this way, make it more clear and easier to maintain.
And, i guess, faster too.

What do you think?

Comment (tcelestino) [reply components/Accordion/Accordion.jsx:32]: Can you use the parents to selector element in unit test? Use styled-components to create element without properties it's wrong for me. 🍡

**Comment by tcelestino** [comment]
accordion-behavior

@rapahaeru, is it normal?  :suspect:

File: components/Accordion/Accordion.jsx (hunk 1 above)
Comment (rapahaeru) [reply components/Accordion/Accordion.jsx:32]: Agree. I fixed changed the styled element to a css class.

**Review by tcelestino — DISMISSED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review
- [x] TypeScript updated
- [x] a11y

### Browsers review
- [x] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [ ] Safari
  - [ ] Mobile
- [ ] Ux review validation

**Comment by toncabral88** [comment]
Screenshot from 2021-04-28 16-14-28

Screenshot from 2021-04-28 16-09-25

The content of accordion item is in a separated container of header, but in the style guide is not. Should it be like that?

**Comment by toncabral88** [comment]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review
- [x] TypeScript updated
- [x] a11y

### Browsers review
- [x] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [x] Safari
  - [x] Mobile
- [ ] Ux review validation

File: components/Accordion/Accordion.jsx [hunk 2]
```diff
@@ -99,13 +105,46 @@ const AccordionContent = styled.div`
...
+  &:first-child {
+    border-radius: ${opened ? '4px' : '4px 4px 0px 0px'};
+  }
+
+  &:last-child {
+    border-radius: ${opened ? '4px' : '0px 0px 4px 4px'};
+    border-bottom: none;
+  }
+
+  border-left: ${opened ? `4px solid ${primary[700]}` : `inherit`};
```
Comment (toncabral88) [inline components/Accordion/Accordion.jsx:140]: I think when the state is closed border left should be same, but with background color. Currently the header jumps a bit for the right when the state changes.

**Review by toncabral88 — DISMISSED** [review]

**Review by toncabral88 — DISMISSED** [review]

File: components/Accordion/Accordion.jsx [hunk 3]
```diff
@@ -222,15 +271,20 @@ Accordion.defaultProps = {
...
+  /** Theme natively passed by Quantum theme. Don't change this prop */
   theme: PropTypes.shape({
     colors: PropTypes.object,
     spacing: PropTypes.object,
     baseFontSize: PropTypes.number,
   }),
   /** Set true to keep only one item open and close the remainings on click */
   keepOnlyOneOpen: PropTypes.bool,
-  /** Items to render in accordion */
+  /**
```
Comment (rodrigoclemente) [inline components/Accordion/Accordion.jsx:282]: Why?

**Comment by rodrigoclemente** [comment]
Could have an animation to be more user friendly

File: components/Accordion/Accordion.jsx (hunk 3 above)
Comment (rapahaeru) [reply components/Accordion/Accordion.jsx:282]: this comments in jsx are used in quantum-storytook-ui to render a explanation of the props use

File: components/Accordion/Accordion.jsx [hunk 4]
```diff
@@ -67,26 +66,30 @@ const AccordionContent = styled.div`
...
     opened,
   }) => `
-    height: ${opened ? 'auto' : '0'};
-    transform: scaleY(${opened ? '1' : '0'});
-    transform-origin: top;
-    overflow: hidden;
-    transition: transform 0.2s ease;
     font-size: ${baseFontSize - 2}px;
-
+    max-height: ${opened ? '2000px' : '0'};
```
Comment (rodrigoclemente) [inline components/Accordion/Accordion.jsx:74]:
```suggestion
    max-height: ${opened ? 'none' : '0'};
```

File: components/Accordion/Accordion.jsx [hunk 5]
```diff
@@ -67,26 +66,30 @@ const AccordionContent = styled.div`
...
+    overflow: hidden;
+    padding: 0px;
+    transition: max-height 0.2s ease;
```
Comment (rodrigoclemente) [inline components/Accordion/Accordion.jsx:77]:
```suggestion
    transition: max-height 0.2s ease-in-out;
```

File: components/Accordion/Accordion.jsx (hunk 4 above)
Comment (rodrigoclemente) [reply components/Accordion/Accordion.jsx:74]: Is better than height limit

File: components/Accordion/Accordion.jsx [hunk 6]
```diff
@@ -67,26 +66,30 @@ const AccordionContent = styled.div`
...
+
```
Comment (rodrigoclemente) [inline components/Accordion/Accordion.jsx:78]:
```suggestion
    will-change: transform;
    
```

Comment (rodrigoclemente) [reply components/Accordion/Accordion.jsx:78]: To use GPU to render animation

File: components/Accordion/Accordion.jsx [hunk 7]
```diff
@@ -96,16 +99,82 @@ const AccordionContent = styled.div`
...
+    margin-right: ${large}px;
+    color: ${primary[700]};
   `}
+
   line-height: inherit;
 `;
 
-AccordionItem.displayName = 'AccordionItem';
+const AccordionItem = styled.li`
+  transition: border 0.2s ease;
```
Comment (rodrigoclemente) [inline components/Accordion/Accordion.jsx:132]:
```suggestion
  transition: border 0.2s ease-in-out;
```

**Review by rodrigoclemente — DISMISSED** [review]

File: components/Accordion/Accordion.jsx [hunk 8]
```diff
@@ -67,26 +66,31 @@ const AccordionContent = styled.div`
...
     opened,
   }) => `
-    height: ${opened ? 'auto' : '0'};
-    transform: scaleY(${opened ? '1' : '0'});
-    transform-origin: top;
-    overflow: hidden;
-    transition: transform 0.2s ease;
     font-size: ${baseFontSize - 2}px;
-
+    max-height: ${opened ? '2000px' : '0'};
```
Comment (toncabral88) [inline components/Accordion/Accordion.jsx:74]: This conditional could be removed and values included in its states

File: components/Accordion/Accordion.jsx [hunk 9]
```diff
@@ -96,16 +100,82 @@ const AccordionContent = styled.div`
...
+    margin-right: ${large}px;
+    color: ${primary[700]};
   `}
+
   line-height: inherit;
 `;
 
-AccordionItem.displayName = 'AccordionItem';
+const AccordionItem = styled.li`
+  transition: border 0.2s ease-in-out;
```
Comment (toncabral88) [inline components/Accordion/Accordion.jsx:133]: I think margin could be included in transition to improve the animation

**Review by toncabral88 — APPROVED** [review]

**Pull request merged**